### PR TITLE
Created a metadata rule to replace composed-jar attribute by jar.

### PR DESCRIPTION
**Pull request by AlexanderBartash**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -33,7 +40,34 @@ tasks.withType(Test) {
...
+//    @Inject abstract ObjectFactory getObjects()
+//
+//    void execute(ComponentMetadataContext context) {
+//        context.details.withVariant("runtimeElements") {
+//            attributes {
+//                attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements, LibraryElements.JAR))
+//            }
+//        }
+//    }
+//}
```
Comment (AlexanderBartash) [inline build.gradle:55]: I created two versions, feel free to use whatever you like.

File: build.gradle [hunk 2]
```diff
@@ -33,7 +40,34 @@ tasks.withType(Test) {
...
+            withVariant("intellijPlatformComposedJar") {
+                attributes {
+                    attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements, LibraryElements.JAR))
+                }
+            }
+        }
+    }
+
     implementation(
             "com.redhat.devtools.intellij:intellij-common:${intellijCommonVersion}",
```
Comment (AlexanderBartash) [inline build.gradle:72]: I just noticed another api which might work you can try:
```kotlin
implementation("com.redhat.devtools.intellij:intellij-common:${intellijCommonVersion}") {
        attributes {
            attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named("composed-jar"))
        }
}
```

**Comment by sbouchet** [comment]
@AlexanderBartash whoo thanks for the PR ! 
we're currenlty migrating our old groovy builds to Kotlin, to be also able to target 2024.2 + .
see #780 with your patch in use

File: build.gradle (hunk 2 above)
Comment (adietish) [reply build.gradle:72]: tested it in a different PR where I needed this: https://github.com/redhat-developer/intellij-kubernetes/pull/781/commits/df398b441f6b4ba5ef6b050452281101bd4b2459 Works super nicely. I'll add you as co-author to my PR. Thanks!

**Comment by adietish** [comment]
Using this in https://github.com/redhat-developer/intellij-kubernetes/pull/781/commits/df398b441f6b4ba5ef6b050452281101bd4b2459. Closing this PR.

**Pull request closed without merging**